website/src/blogs/openebs-node-device-management-ndm-e2-80-8a--e2-80-8atroubleshooting-tips.md

```diff
@@ -0,0 +1,120 @@
+
+---
+
+### NDM Quick Overview
+
+For setting up NDM in secure mode, please see my previous [blog](https://blog.mayadata.io/openebs/configuring-openebs-to-run-with-security-enhanced-linux), and you can learn how NDM works [here](https://docs.openebs.io/docs/next/ndm.html). Here is a quick snapshot of the key components of NDM.
+
+- NDM components are installed in the OpenEBS Namespace. Ensure that NDM pods part of the NDM DaemonSet are running on all the storage nodes. NDM Operator helps with allocating Block Devices to Block Device Claims and should be running.
+- NDM DaemonSet pod discovers all the block devices attached to the node and creates BlockDevice custom resource for each device. Note that NDM will filter out some of the devices like loopback device and so forth as configured in the NDM ConfigMap. `kubectl get bd -n openebs`
+- NDM creates a special type of devices called sparse devices depending on the `SPARSE_FILE_COUNT` and `SPARSE_FILE_SIZE` passed to the NDM Daemon. These devices are used in cases where nodes do not have any additional devices attached to the node and users would like to run their applications by carving out some spaces from the OS disk. The creation of sparse devices is disabled by default from OpenEBS 1.3.
+- Users or Operators like cStor Operator, Local PV provisioner interact with NDM by creating a BlockDeviceClaim CR. The BlockDeviceClaim will have properties like nodeName, required Capacity etc., The NDM operator will match these properties with the available BlockDevices and associate the one that matches all the requirements to BlockDeviceClaim.
+
+---
+
+### NDM Known Issues / Future Development Items
+
+- BlockDevices are not created for Partitions and LVM devices. If you need to use them, you have to manually create BlockDevice CR. The steps are mentioned in this [blog](https://blog.mayadata.io/openebs/creating-manual-blockdevice).
+
+OK. Let us get started with some common issues reported and how to troubleshoot them.
+
+---
+
+#### Scenario #1
+
+**BlockDevice CR is not created for a device available on my node.**
+
+I have some disks attached to the node. Installed openebs, but blockdevice resources are not created for the devices.
+
+**Symptom:** I have some disks attached to the node. Installed openebs, but blockdevice resources are not created for the devices.
+
+**Troubleshooting:**
+
+1. Check `lsblk` output of the node
+2. Get the NDM config map.
+3. Check if the mount point of the disk is excluded in the filter configurations in configmap.
+4. From lsblk output check if the blockdevice you want to use is an LVM/software raid/ partition/LUKS filesystem. NDM currently does not support these types.
+5. If none of the above works, the logs of NDM daemonset can be checked. It will have information of disk being detected , and at what point the disk was excluded from blockdevice creation, (like `excluded by path-filter`)
```

Comment:
```suggestion
5. If none of the above works, the logs of NDM daemonset can be checked. It will have information of disk being detected, and at what point the disk was excluded from blockdevice creation, (like `excluded by path-filter`)
```